rhel/vex/parser.go

```diff
@@ -362,9 +385,58 @@ func (c *creator) lookupVulnerability(vulnKey string, protoVulnFunc func() *clai
 	return &c.fixedVulns[idx], ok
 }
 
+type ranger struct {
+	lowest map[string]*claircore.Range
+}
+
+func newRanger() *ranger {
+	return &ranger{
+		lowest: map[string]*claircore.Range{},
+	}
+}
+
+// Add takes a packageName and a fixedInVersion and returns a *claircore.Range relating
+// to the fixedInVersion. Add also saves the lowest range per packageName that it's seen.
+// This allows resetLowest() to zero out the lowest values we saw per packageName.
+func (r *ranger) add(packageName, fixedInVersion string) (*claircore.Range, error) {
+	var rng = &claircore.Range{}
+
+	if fixedInVersion == "" {
+		zeroVer := &rhctag.Version{}
+		rng.Lower = zeroVer.Version(true)
+		highestVer := rhctag.Version{
+			Major: math.MaxInt32, // Everything is vulnerable
+		}
+		rng.Upper = highestVer.Version(true)
+	} else {
+		firstPatch, err := rhctag.Parse(fixedInVersion)
+		if err != nil {
+			return nil, err
+		}
+		rng.Upper = firstPatch.Version(false)
```

Comment:
this set the range to something like 4.5.0 - 4.5.MAX_INT. Shouldn't we only be affected up to the fixedInVersion, though?